src/main/java/org/folio/qm/domain/repository/RecordCreationStatusRepository.java

```diff
@@ -3,12 +3,29 @@
 import java.util.Optional;
 import java.util.UUID;
 
+import javax.transaction.Transactional;
+
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import org.folio.qm.domain.entity.RecordCreationStatus;
+import org.folio.qm.domain.entity.RecordCreationStatusUpdate;
 
 public interface RecordCreationStatusRepository extends JpaRepository<RecordCreationStatus, UUID> {
 
   Optional<RecordCreationStatus> findByJobExecutionId(UUID jobExecutionId);
 
+  @Modifying
+  @Transactional
+  @Query("UPDATE RecordCreationStatus s SET "
+    + "s.status =:#{#update.status}, "
+    + "s.instanceId =:#{#update.instanceId}, "
+    + "s.marcBibId =:#{#update.marcBibId}, "
+    + "s.errorMessage =:#{#update.errorMessage} "
+    + "WHERE s.jobExecutionId =:jobExecutionId"
+  )
+  int update(@Param("update") RecordCreationStatusUpdate update, @Param("jobExecutionId") UUID jobExecutionId);
```

Comment:
Does this change somehow relate to DB connection issue or it's an outside improvement

Reply:
This change will decrease the number of requests to DB that will lead to connections will be released quicker.

Reply:
well, an additional question then: have you noticed during the investigation that with the previous version, which had I believe 2 requests to DB (a select and an update), there were some long running sql requests that might cause a connection to be occupied for a long period of time?